### PR TITLE
TOMEE-4424 - improved JAX-RS request matching if a welcome-file is defined

### PR DESCRIPTION
**Pull request by jungm**
See https://issues.apache.org/jira/browse/TOMEE-4424

Fulll build is running: https://ci-builds.apache.org/job/Tomee/view/tomee-10.x/job/pull-request-manual/146/

### CODE REVIEW

File: arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/MyFiletypeServlet.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.apache.openejb.arquillian.tests.jaxrs.servlets.welcomefile;
```
Comment (rzo1) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/MyFiletypeServlet.java:1]: Header

File: arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/MyRestApi.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.apache.openejb.arquillian.tests.jaxrs.servlets.welcomefile;
```
Comment (rzo1) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/MyRestApi.java:1]: Header

File: arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/MyRestApplication.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.apache.openejb.arquillian.tests.jaxrs.servlets.welcomefile;
```
Comment (rzo1) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/MyRestApplication.java:1]: Header

File: arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/ServletsWithWelcomeFileTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
+package org.apache.openejb.arquillian.tests.jaxrs.servlets.welcomefile;
```
Comment (rzo1) [inline arquillian/arquillian-tomee-tests/arquillian-tomee-jaxrs-tests/src/test/java/org/apache/openejb/arquillian/tests/jaxrs/servlets/welcomefile/ServletsWithWelcomeFileTest.java:1]: Header

**Comment by jungm** [comment]
Ran the JAX-RS TCK now:
```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   JAXRSClientIT.getMediaTypeTest:129->ReaderClient.invoke:84 » Fault jakarta.ws.rs.client.ResponseProcessingException: Problem with reading the data, class java.lang.String, ContentType: application/json.
[ERROR]   JAXRSClientIT.getMediaTypeTest:149->WriterClient.invoke:57 » Fault ee.jakarta.tck.ws.rs.common.JAXRSCommonClient$Fault: [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.getMediaTypeTest:180->ReaderClient.invoke:50->ReaderClient.invoke:64 » Fault ee.jakarta.tck.ws.rs.common.JAXRSCommonClient$Fault: [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.getMediaTypeTest:140->WriterClient.invoke:39->JaxrsCommonClient.invoke:209 » ResponseProcessing Problem with reading the data, class java.lang.String, ContentType: application/json.
[ERROR]   JAXRSClientIT.invocationFromLinkApplicationJsonMediaTypeTest:154->checkResposeForMessage:194 » ResponseProcessing Problem with reading the data, class java.lang.String, ContentType: application/json.
[ERROR]   JAXRSClientIT.byteArrayProviderTest:172->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.dataSourceProviderTest:260->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.fileProviderTest:243->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.inputStreamProviderTest:206->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.readerProviderTest:226->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.streamingOutputProviderTest:295->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[ERROR]   JAXRSClientIT.stringProviderTest:189->setEntityAndPropertyAndInvoke:73->setPropertyAndInvoke:66->JAXRSCommonClient.invoke:328 » Fault [JAXRSCommonClient] null failed!  Check output for cause of failure.
[INFO] 
[ERROR] Tests run: 2785, Failures: 0, Errors: 12, Skipped: 132
```

This did definitely break something, I'll look at this

**Comment by rzo1** [comment]
Let's see if the adaptions in https://github.com/apache/tomee/tree/cxf-4.1.x-jsonp-fixes will fix the TCK/build on the manual full build, so we get it sorted out.

**Comment by rzo1** [comment]
So let's see how it goes https://ci-builds.apache.org/job/Tomee/job/pull-request-manual/153/

**Comment by jungm** [comment]
the 2 failing tests are unrelated, see https://github.com/apache/tomee/commit/9a68d9f8f775be4996a2006abe31de2cb3e02f8c

**Pull request merged**